diff --git src/main/java/com/google/devtools/build/lib/util/CPU.java src/main/java/com/google/devtools/build/lib/util/CPU.java
index 7a85c29..157119d 100644
--- src/main/java/com/google/devtools/build/lib/util/CPU.java
+++ src/main/java/com/google/devtools/build/lib/util/CPU.java
@@ -25,7 +25,7 @@ public enum CPU {
   X86_32("x86_32", ImmutableSet.of("i386", "i486", "i586", "i686", "i786", "x86")),
   X86_64("x86_64", ImmutableSet.of("amd64", "x86_64", "x64")),
   PPC("ppc", ImmutableSet.of("ppc", "ppc64", "ppc64le")),
-  ARM("arm", ImmutableSet.of("arm", "armv7l")),
+  ARM("arm", ImmutableSet.of("aarch64", "arm", "armv7l" )),
   S390X("s390x", ImmutableSet.of("s390x", "s390")),
   UNKNOWN("unknown", ImmutableSet.<String>of());
 
